core/src/main/java/com/netflix/conductor/core/events/SimpleActionProcessor.java

```diff
@@ -123,6 +127,23 @@ public SimpleActionProcessor(
                 return replaced;
             }
             taskModel = workflow.getTaskByRefName(taskRefName);
+            // Task can be loopover task.In such case find corresponding task and update
+            List<TaskModel> loopOverTaskList =
+                    workflow.getTasks().stream()
+                            .filter(
+                                    t ->
+                                            TaskUtils.removeIterationFromTaskRefName(
+                                                            t.getReferenceTaskName())
+                                                    .equals(taskRefName))
+                            .collect(Collectors.toList());
+            if (!loopOverTaskList.isEmpty()) {
+                // Find loopover task with the highest iteration value
```

Comment:
The assumption here is loop over task with the highest iteration only will be waiting for an event. Since older iteration gets completed.